php/docker/templates/nginx-vhost-conf.d/80-index.conf

```diff
@@ -1,10 +1,118 @@
 # vim: set ft=nginx:
 
+{{- $stackBackends := list "redis" "memcached" "custom" }}
+{{- $truthStrings := list "true" "on" "yes" -}}
+{{- $keyPrefix := default "nginx-cache:" .Env.STACK_PAGE_CACHE_KEY_PREFIX }}
+{{- $keyUID := default "https$request_method$host$request_uri" .Env.STACK_PAGE_CACHE_KEY_UID }}
+
 location / {
     try_files $uri $uri/ /index.php$is_args$args;
 }
 
 location ~ \.php$ {
+    {{- if has .Env.STACK_PAGE_CACHE_BACKEND $stackBackends }}
+    set $skip_cache 0;
+
+    # POST requests and urls with a query string should always go to PHP
+    if ($request_method = POST) {
+        set $skip_cache 1;
+    }
+    if ($query_string != "") {
+        set $skip_cache 1;
+    }
+
+    # Don't cache uris containing the following segments
+    if ($request_uri ~* "/wp-admin/|/xmlrpc.php|wp-.*.php|/feed/|index.php|sitemap(_index)?.xml") {
```

Comment:
It should work for multisites, right?

Reply:
It needs to be tested in that scenario to know for sure.